src/adera/Machines/Rocket.cpp

```diff
@@ -163,9 +170,86 @@ void SysMachineRocket::update_physics()
     }
 }
 
+void SysMachineRocket::attach_plume_effect(ActiveEnt ent)
+{
+    using Magnum::GL::Mesh;
+    using Magnum::GL::Texture2D;
+    using Magnum::Trade::ImageData2D;
+    using namespace Magnum::Math::Literals;
+
+    ActiveEnt plumeNode = entt::null;
+
+    auto findPlumeHandle = [&](ActiveEnt ent)
+    {
+        ACompHierarchy& node = m_scene.reg_get<ACompHierarchy>(ent);
+        if (node.m_name.compare(0, 9, "fx_plume_") == 0)
+        {
+            plumeNode = ent;
+            return false;  // terminate search
+        }
+        return true;
+    };
+
+    m_scene.hierarchy_traverse(ent, findPlumeHandle);
```

Comment:
Instead of making your functor return a true or false, which is convienient, i know, define an enum class. E.g.
enum class EHierarchyTraverseStatus : bool
{
    Continue = true,
    Stop = false
};
(Or use uint8_t and 0, 1, if that doesn't compile)

Additionally, consider changing your hierarchy_traverse function to return decltype(auto), and have it return the result of your functor, that would let you change your lambda to return `{EHierarchyTraverseStatus::Stop, ent}` or `{EHierarchyTraverseStatus::Continue, entt::null}`

Reply:
Searching for a certain entity is just one of many things you could do with this function; it could be for any process that needs to recursively touch all the nodes. Or am I misunderstanding the point of your last sentence?

Reply:
Yes, but the magic of templates :-)

If you write your function carefully (Or your lambda, depending on how crafty your metaprogramming game is) your hierarchy_traverse() function can serve double duty.

In the lazy way, you just always have your hierarchy_traverse() function return the second member of the thing returned from the lambda, and for lambdas that don't want to return anything, just have them return like nullptr or something.

In the crafty way, you have your hierarchy_traverse function return a tuple that represents all of the return values from the lambda except for index 0 (e.g. a tuple suffix).

If you write your hierarchy_traverse function carefully, there's very little cost to doing this. E.g. returning a tuple<EHierarchyTraverseStatus, std::reference_wrapper<>, std::reference_wrapper<>,....>, where the cost is basically just shuffling pointers around, which the compiler will heavily optimize for you cause of templates.